### PR TITLE
Added --ci flag for continuous integration

### PR DESCRIPTION
**Pull request by timneutkens**
All commands now support --ci and will fail on error when it's provided. Fixes https://github.com/SnowdogApps/magento2-frontools/issues/24 in this repository for now. @Igloczek you were planning on porting this to frontools right?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
     "gulp-sass": "^2.3.2",
+    "gulp-sass-error": "^1.0.1",
     "gulp-sass-lint": "^1.1.1",
     "gulp-util": "^3.0.7",
     "postcss-reporter": "^1.3.3",
     "run-sequence": "^1.2.1",
     "stylelint": "^6.6.0"
+  },
+  "dependencies": {
+    "gulp-sass-error": "^1.0.2"
```
Comment (Igloczek) [inline package.json]: Let's add this to `devDependencies`

Comment (timneutkens) [inline package.json]: Ah, good catch. Will move that indeed.

File: gulpfile.js [hunk 2]
```diff
@@ -39,49 +44,21 @@ gulp.task('sass', () => {
...
                 clearMessages: true,
-                throwError   : true
+                throwError: config.ci
             })
         ]));
 });
 
 gulp.task('ci-tests', () => {
-    runSequence('ci:sass-lint', 'ci:sass', 'ci:css-lint');
+    runSequence('sass-lint', 'sass', 'css-lint');
```
Comment (Igloczek) [inline gulpfile.js:63]: Let's set here `config.ci = true;`

File: .travis.yml [hunk 3]
```diff
@@ -3,4 +3,4 @@ node_js: 4
 before_script:
   - npm install -g gulp
 script:
-  - gulp ci-tests
+  - gulp ci-tests --ci
```
Comment (Igloczek) [inline .travis.yml]: It's a little bit weird to set `ci` flag to `ci-tests` 😉

Comment (timneutkens) [inline .travis.yml]: yeah :wink: will remove it when `config.ci = true` is added.

**Comment by Igloczek** [comment]
Looks like CI tests are failing now:

```
TypeError: sassError is not a function
```

File: .travis.yml (hunk 3 above)
Comment (timneutkens) [inline .travis.yml]: Or rename ci-tests to just tests. So you can use `gulp tests` for testing locally and `gulp tests --ci` to test using travis. What would you prefer?

Comment (Igloczek) [inline .travis.yml]: Second option, with flag is better.

**Comment by timneutkens** [comment]
Made the changes we talked about. Also fixed the build.

**Pull request merged**